web/pgadmin/browser/__init__.py

```diff
@@ -724,18 +724,19 @@ def set_master_password():
 
         try:
             all_server = Server.query.all()
+            saved_password_servers = [server for server in all_server if
+                                      server.save_password]
             # pgAdmin will use the OS password manager to store the server
             # password, here migrating the existing saved server password to
             # OS password manager
-            if keyring.get_password(
+            if len(saved_password_servers) > 0 and keyring.get_password(
```

Comment:
so even if one server has saved password then it will call get password for all?

Reply:
No for all, it is for specific server only.